Demo/src/main/java/com/braintreepayments/demo/SEPADirectDebitFragment.java

```diff
@@ -74,24 +78,48 @@ public void launchSEPADirectDebit(View view) {
 
         sepaDirectDebitClient.createPaymentAuthRequest(request, (paymentAuthRequest) -> {
             if (paymentAuthRequest instanceof SEPADirectDebitPaymentAuthRequest.Failure) {
-                handleError(((SEPADirectDebitPaymentAuthRequest.Failure) paymentAuthRequest).getError());
+                handleError(
+                        ((SEPADirectDebitPaymentAuthRequest.Failure) paymentAuthRequest).getError());
             } else if (paymentAuthRequest instanceof SEPADirectDebitPaymentAuthRequest.LaunchNotRequired) {
-                handleSEPANonce(((SEPADirectDebitPaymentAuthRequest.LaunchNotRequired) paymentAuthRequest).getNonce());
+                handleSEPANonce(
+                        ((SEPADirectDebitPaymentAuthRequest.LaunchNotRequired) paymentAuthRequest).getNonce());
             } else if (paymentAuthRequest instanceof SEPADirectDebitPaymentAuthRequest.ReadyToLaunch) {
-                sepaDirectDebitLauncher.launch(requireActivity(), (SEPADirectDebitPaymentAuthRequest.ReadyToLaunch) paymentAuthRequest);
+                SEPADirectDebitPendingRequest pendingRequest =
+                        sepaDirectDebitLauncher.launch(requireActivity(),
+                                (SEPADirectDebitPaymentAuthRequest.ReadyToLaunch) paymentAuthRequest);
+                if (pendingRequest instanceof SEPADirectDebitPendingRequest.Started) {
+                    this.pendingRequest = (SEPADirectDebitPendingRequest.Started) pendingRequest;
+                } else {
+                    handleError(
+                            ((SEPADirectDebitPendingRequest.Failure) pendingRequest).getError());
+                }
+            }
+        });
+    }
+
+    private void completeSEPAFlow(SEPADirectDebitPaymentAuthResult paymentAuthResult) {
+        sepaDirectDebitClient.tokenize(paymentAuthResult, (result) -> {
+            if (result instanceof SEPADirectDebitResult.Failure) {
+                handleError(((SEPADirectDebitResult.Failure) result).getError());
+            } else if (result instanceof SEPADirectDebitResult.Cancel) {
+                handleError(new UserCanceledException("User canceled SEPA Direct Debit"));
+            } else if (result instanceof SEPADirectDebitResult.Success) {
+                handleSEPANonce(((SEPADirectDebitResult.Success) result).getNonce());
             }
         });
     }
 
     private String generateRandomCustomerId() {
-        return UUID.randomUUID().toString().substring(0,20);
+        return UUID.randomUUID().toString().substring(0, 20);
```

Comment:
does a linter yell at this, is that why we added the space?

Reply:
Yes this was a result of using Android Studio's auto reformat code tool (mostly breaks up newlines where necessary).